dpreference/src/main/java/me/dozen/dpreference/PreferenceProvider.java

```diff
@@ -79,33 +90,30 @@ public Cursor query(Uri uri, String[] projection, String selection, String[] sel
                     cursor = preferenceToCursor(getDPreference(model.getName()).getPrefLong(model.getKey(), -1));
                 }
                 break;
+            default:
+                // do nothing
         }
         return cursor;
     }
 
-    @Nullable
     @Override
     public String getType(Uri uri) {
         return null;
     }
 
-    @Nullable
     @Override
-    public Uri insert(Uri uri, ContentValues values) {
+    public int insert(Uri uri, ValuesBucket values) {
         throw new IllegalStateException("insert unsupport!!!");
     }
 
     @Override
-    public int delete(Uri uri, String selection, String[] selectionArgs) {
-        switch (sUriMatcher.match(uri)) {
+    public int delete(Uri uri, DataAbilityPredicates predicates) {
+        String lastPath = PreferenceProvider.changeParamToPath(uri);
+        switch (getPathId(lastPath)) {
             case PREF_BOOLEAN:
             case PREF_LONG:
             case PREF_STRING:
             case PREF_INT:
```

Comment:
Ideally, the preference must be deleted when delete api is called. But here removePreference implementation is missing in this delete api method. Please check.

Reply:
Done